packages/amplify-category-function/src/provider-utils/awscloudformation/utils/consolidateDependsOn.ts

```diff
@@ -0,0 +1,30 @@
+import { $TSObject } from 'amplify-cli-core';
+import { FunctionDependency } from 'amplify-function-plugin-interface';
+import _ from 'lodash';
+import { category } from '../../../constants';
+import { lambdaLayerSetting, resourceAccessSetting, ServiceName } from './constants';
+
+export const consolidateDependsOnForLambda = (
```

Comment:
What about cron settings, do we need to add handling for those as well?

Reply:
Nope, we only have conflicts in dependsOn{resourceaccess , lambdalayer} , since its shared among resources. Cron setting only changes functionParameters:{cloudwatcRule}.In update step ,  if we select cron, then it will chnage the above parameter keeping previous dependsOn intact.